llvm/lib/CodeGen/RegAllocGreedy.h

```diff
@@ -183,6 +186,25 @@ class LLVM_LIBRARY_VISIBILITY RAGreedy : public MachineFunctionPass,
   Optional<ExtraRegInfo> ExtraInfo;
   std::unique_ptr<RegAllocEvictionAdvisor> EvictAdvisor;
 
+  std::unique_ptr<MLModelRunner> Evaluator;
+  // std::unique_ptr<Logger> Log;
+  Logger *Log = nullptr;
+  StringMap<std::unique_ptr<Logger>> LogMap;
+  std::string LogPath;
+
+  bool doFinalization(Module &M) override {
+    if (LogPath.empty())
```

Comment:
if you move the implementation to the .cpp, you don't need the LogPath variable, just pick it up from the command line option variable.